semanticdb/semanticdb/semanticdb.proto

```diff
@@ -157,13 +140,37 @@ message RepeatedType {
   Type tpe = 1;
 }
 
-message TypeType {
-  reserved 1;
-  Scope type_parameters = 4;
+message Signature {
+  oneof sealed_value {
+    ClassSignature classSignature = 1;
+    MethodSignature methodSignature = 2;
+    TypeSignature typeSignature = 3;
+    ValueSignature valueSignature = 4;
+  }
+}
+
+message ClassSignature {
+  Scope type_parameters = 1;
+  repeated Type parents = 2;
+  Scope declarations = 3;
+}
+
+message MethodSignature {
+  Scope type_parameters = 1;
+  repeated Scope parameterLists = 2;
+  Type return_type = 3;
+}
+
+message TypeSignature {
+  Scope type_parameters = 1;
   Type lower_bound = 2;
   Type upper_bound = 3;
 }
 
+message ValueSignature {
```

Comment:
Can you explain the motivation for the name "ValueSignature" as compared to for example "FieldSignature"? I like "ValueSignature", but it took a moment for me to click.

Reply:
SLS separates value types and non-value types (i.e. things like signatures). `ValueSignature` is inspired by this separation - it's for all definitions whose signature is a value type. Just `FieldSignature` wouldn't be enough, because it also applies to parameters, self parameters and locals.